sdk/ml/azure-ai-ml/azure/ai/ml/operations/_job_operations.py

```diff
@@ -1150,7 +1150,7 @@ def _resolve_arm_id_for_pipeline_job(self, pipeline_job: "PipelineJob", resolver
             and getattr(pipeline_job.component, "_source", None) == ComponentSource.YAML_COMPONENT
         ):
             pipeline_job.component = resolver(
-                pipeline_job._to_component(silent=True),
```

Comment:
this is a bugfix? what's the difference?

Reply:
When source is YAML_Component. the registered component should honor what we loaded from file(i.e. pipeline_job.component), the pipeline_job._to_component will build component input type base on job's real input value, this is not as expected.